### DISCUSSION TITLE
useFieldArray not accepting object as value

### DISCUSSION THREAD

**Discussion by LauSam09** [react-hook-form/react-hook-form#4051] [Q&A] [answered]
I've created a custom input that accepts an object as a value (e.g. a number and a string value). I've wrapped this in `<Controller />` and used it without problems in many places, until I've tried to use it in `useFieldArray`, when I get a warning and bizarre behaviour.

![image](https://user-images.githubusercontent.com/24778857/106275791-54767880-622e-11eb-88f0-dc3432ba9cd4.png)

The accessor here should be the object in the array itself - not one of its properties. The result is that the input is displayed on screen correctly, but the array is not present when the form is submitted.

I've created a minimal code sandbox (just shows displaying of data - no editing) to demonstrate the problem.
https://codesandbox.io/s/react-hook-form-usefieldarray-with-object-oxjrx

Any advice? Thanks!

**Comment by bluebill1049** [comment, marked as answer]
have a read on the important list of things: https://react-hook-form.com/api#useFieldArray

This CSB should be helpful to you: https://codesandbox.io/s/react-hook-form-usefieldarray-vy8fv

**Reply by LauSam09** [reply]
Thanks Bill - from reading that, I think my easiest solution will be to nest the array properties within another property... e.g.

```js
// old
//const array = [{ value: 1, name: "tim" }, { value : 2, name: "bill" } ]
const array = [{ value: { value: 1, name: "tim" }}, { value: { value : 2, name: "bill" }} ]
```

and then the accessor can be `array[index].value` as opposed to `array[index]`.

I've updated the code sandbox and this seems to work!
